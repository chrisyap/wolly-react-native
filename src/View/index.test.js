import React from "react";
import V, { AnimatedV } from "./index";

describe("V", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<V />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders row correctly", () => {
    const props = {
      flexDirection: "row"
    };
    const wrapper = shallow(<V {...props} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders alignments correctly", () => {
    const props = {
      alignItems: "center",
      alignContent: "center",
      justifyContent: "center"
    };
    const wrapper = shallow(<V {...props} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });
});

describe("AnimatedV", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<AnimatedV />);
    expect(wrapper).toMatchSnapshot();
  });
});
