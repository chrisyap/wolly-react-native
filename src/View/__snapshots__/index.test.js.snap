// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedV renders correctly 1`] = `
<StyledNativeComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "inlineStyle": InlineStyle {
        "rules": Array [
          "
  ",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          ";
",
        ],
      },
      "render": [Function],
      "shouldForwardProp": undefined,
      "styledComponentId": "StyledNativeComponent",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`V renders alignments correctly 1`] = `
<View
  alignContent="center"
  alignItems="center"
  justifyContent="center"
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      },
    ]
  }
/>
`;

exports[`V renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
/>
`;

exports[`V renders row correctly 1`] = `
<View
  flexDirection="row"
  style={
    Array [
      Object {
        "alignContent": "flex-start",
        "alignItems": "stretch",
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
    ]
  }
/>
`;
